packages/mwp-router/src/util/matcher.js

```diff
@@ -0,0 +1,74 @@
+// @flow
```

Comment:
this module is a little refactor of some things that were in `resolve.js` - it includes _synchronous_ functions that return `route` objects that match a passed-in location/URL. The previous implementations were async functions that used `getNestedRoutes`/`getIndexRoute`, but those features are gone so we can do things more simply. There's still some recursion, but it should be more clear - please ask questions if the docs are insufficient!